File: android/app/src/main/java/com/lilru/liftr/nutrition/NutritionLabelSpatialTypes.kt
package com.lilru.liftr.nutrition

data class NutritionLabelSpatialElement(
    val text: String,
    val minX: Float,
    val minY: Float,
    val maxX: Float,
    val maxY: Float
) {
    val centerX: Float get() = (minX + maxX) / 2f
    val centerY: Float get() = (minY + maxY) / 2f
    val rightX: Float get() = maxX
}

data class NutritionLabelRecognitionResult(
    val mergedLines: List<String>,
    val elements: List<NutritionLabelSpatialElement>
)

File: Liftr/NutritionLabelSpatialTypes.swift
import CoreGraphics
import Foundation

struct NutritionLabelSpatialElement: Equatable {
    let text: String
    let minX: CGFloat
    let minY: CGFloat
    let maxX: CGFloat
    let maxY: CGFloat

    var centerX: CGFloat { (minX + maxX) / 2 }
    var centerY: CGFloat { (minY + maxY) / 2 }
    var rightX: CGFloat { maxX }
    var width: CGFloat { maxX - minX }
    var height: CGFloat { maxY - minY }
}

struct NutritionLabelRecognitionResult: Equatable {
    let mergedLines: [String]
    let elements: [NutritionLabelSpatialElement]
}
